controller/common/manage_children.go

```diff
@@ -103,6 +103,9 @@ func ManageChildren(dynClient *dynamicclientset.Clientset, updateStrategy ChildU
 func deleteChildren(client *dynamicclientset.ResourceClient, parent *unstructured.Unstructured, observed, desired map[string]*unstructured.Unstructured) error {
 	var errs []error
 	for name, obj := range observed {
+		if parent.GetNamespace() == "" && obj.GetNamespace() != "" {
+			client = client.WithNamespace(obj.GetNamespace())
```

Comment:
I just noticed this is reassigning to the top-level `client` var on each iteration, which technically works (it's like chaining `client.WithNamespace(a).WithNamespace(b)`) but may be confusing to read later. Also, I think it's fine to just call `WithNamespace()` all the time, since we trust the namespace value in the observed object metadata.

That means this can be simplified to just replacing `client.Delete` below with `client.WithNamespace(obj.GetNamespace()).Delete`.

Reply:
Good point, I think I'll update the function docs.